### PR TITLE
improvement: MongoDB - JSON syntax error sent back to Client

### PR DESCRIPTION
**Pull request by agrawaltejas01**
## What type of PR is this? 

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] New Query Runner (Data Source) 
- [ ] New Alert Destination
- [ ] Other

## How is this tested?

- [ ] Unit tests (pytest, jest)
- [ ] E2E Tests (Cypress)
- [ ] Manually
- [x] N/A

### CODE REVIEW

**Comment by agrawaltejas01** [comment]
[Screenshot of Client](https://drive.google.com/file/d/1KMUFJP_FhnUX-Q7iR3Td9qMtLdRL_udP/view?usp=share_link)

**Comment by Avey777** [comment]
It is already difficult to merge the code here. You have another option available:
https://github.com/RedashCommunity/redash

File: redash/query_runner/mongodb.py [hunk 1]
```diff
@@ -266,8 +266,8 @@ def run_query(self, query, user):
 
         try:
             query_data = parse_query_json(query)
-        except ValueError:
-            return None, "Invalid query format. The query is not a valid JSON."
+        except ValueError as error:
+            return None, error.__str__()
```
Comment (guidopetri) [inline redash/query_runner/mongodb.py:270]: We might want to keep the old message and just append the error message that is returned.

**Comment by guidopetri** [comment]
@agrawaltejas01 thanks for your contribution! Would you mind rebasing on `master` and pushing your branch again to re-run the CI checks?

File: redash/query_runner/mongodb.py [hunk 2]
```diff
@@ -267,8 +267,8 @@ def run_query(self, query, user):
...
```
Comment (junnplus) [inline redash/query_runner/mongodb.py:271]:
```suggestion
            return None, f"Invalid JSON format. {error.__str__()}"
```

**Review by junnplus — CHANGES_REQUESTED** [review]

File: redash/query_runner/mongodb.py (hunk 2 above)
Comment (justinclift) [reply redash/query_runner/mongodb.py:271]: That sounds like a good idea. :smile:

**Comment by justinclift** [comment]
Something bizarre there.  The Codecov test when it was merged into `master` shows as a failure, even though when looking at the test details it clearly says there was no change in coverage (0.00%).

It *seems* like the Codecov bot is sometimes flaky.  That's not good. :frowning:

**Comment by junnplus** [comment]
I may know the reason. Sometimes the master's test coverage results are not uploaded successfully, resulting in incorrect test coverage in the next merge report.

**Comment by justinclift** [comment]
Interesting.  If it happens more than rarely, we might need to add some kind of "retry" attempt to the upload then. :smile:

**Pull request merged**